### PR TITLE
Improve liquid staking

### PR DESCRIPTION
**Pull request by MrPai**
add approved agent list and add relaychain era_index

### CODE REVIEW

File: runtime/parallel/src/lib.rs [hunk 1]
```diff
@@ -361,6 +361,24 @@ impl pallet_loans::Config for Runtime {
...
 }
 
+parameter_types! {
+    pub const LiquidStakingAgentMaxMembers: u32 = 100;
+}
+
+type LiquidStakingAgentMembershipInstance = pallet_membership::Instance3;
+impl pallet_membership::Config<LiquidStakingAgentMembershipInstance> for Runtime {
+    type Event = Event;
+    type AddOrigin = EnsureRoot<AccountId>;
```
Comment (GopherJ) [inline runtime/parallel/src/lib.rs:371]: we should also allow council to update

**Comment by GopherJ** [comment]
otherwise it looks fine to me

**Comment by RainFallsSilent** [comment]
I think this pull request should be merged after stake-client is ready to deploy.

**Comment by GopherJ** [comment]
also @MrPai do we have some hooks to use in membership pallet? such as `onRemove` we need hooks to handle the tokens left in agent account.

**Comment by MrPai** [comment]
> also @MrPai do we have some hooks to use in membership pallet? such as `onRemove` we need hooks to handle the tokens left in agent account.

None for now, the agent is a multi-sig account on relaychain, we should confirm that there is no token bonded by it before we remove.  however "bond action" is on relaychain, but "onRemove" is hooked on parachain, there may be some details to design.

I agree this is necessary but should have a clear process before.

**Comment by MrPai** [comment]
> I think this pull request should be merged after stake-client is ready to deploy.

Actually we can merge it now, because current `stake-client` testing is based on a docker-composer file that in another branch, don't worry about the onchain interaction with stake-client.

File: pallets/liquid-staking/src/benchmarking.rs [hunk 2]
```diff
@@ -64,11 +64,12 @@ where
         SystemOrigin::Signed(caller.clone()).into(),
         unstake_amount
     ));
-
+    let era_index = 1;
```
Comment (kaichaosun) [inline pallets/liquid-staking/src/benchmarking.rs:67]: There is no need for this assignment, just use `1` in assert_ok should be fine.

File: pallets/liquid-staking/src/lib.rs [hunk 3]
```diff
@@ -483,6 +501,7 @@ pub mod pallet {
             let new_unstake = UnstakeInfo {
                 amount,
                 block_number,
+                era_index: Some(era_index),
```
Comment (kaichaosun) [inline pallets/liquid-staking/src/lib.rs:504]: Era index should be updated after the offchain staking client process it successfully.

Comment (MrPai) [reply pallets/liquid-staking/src/lib.rs:504]: As I know, the Era index should be fixed after we unbond on relaychain. if we unbond a second time on relaychain, it is totally another Era index.
The era index is associated with the amount each time invoking `unbond`.
please correct me if I have any misunderstandings.

**Review by kaichaosun — APPROVED** [review]
As the `process_pending_unstake` is asynchronous, need a way to identify if it's success or not.

**Pull request merged**